src/daemon/emitter.cc

```diff
@@ -430,15 +447,17 @@ void Emitter::DoRemoteExecute(const base::WorkerPool& pool, ResolveFn resolver,
 
     // Check that we have a compiler of a requested version.
     net::proto::Status status;
-    if (!SetupCompiler(incoming->mutable_flags(), &status)) {
+    const bool has_compiler_path =
```

Comment:
Then why don't you move this logic inside `SetupCompiler()` and handle it properly and transparently? - Otherwise you risk to end up with a different logic in different places.